### PR TITLE
CART-89 mercury: Update to v2.1.0rc4

### PR DESCRIPTION
**Pull request by frostedcmos**
- Update mercury to v2.1.0rc3

Signed-off-by: Alexander Oganezov <alexander.a.oganezov@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: mercury.spec
Comment (brianjmurrell) [inline mercury.spec:16]: Looks like some kind of whitespace issue here you could fix if you need to push again for some reason.

File: mercury.spec
Comment (brianjmurrell) [inline mercury.spec:112]: Add a line documenting the addition of the patch?

File: packaging/Dockerfile.mockbuild
Comment (brianjmurrell) [inline packaging/Dockerfile.mockbuild:9]: We are tracking daos-stack/packaging#60 for packaging at the moment.  That will land to master once it's seen a bit more testing.  But updating packaging here in this PR from packaging's master has rolled back a bunch of needed changes.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Mainly just don't want to back out the packaging/ changes from daos-stack/packaging#60.

File: mercury.spec
Comment (frostedcmos) [reply mercury.spec:16]: This seems to be github rendering issue; shows up aligned in vim/same whitespaces as lines above

File: mercury.spec
Comment (frostedcmos) [reply mercury.spec:112]: will add

File: packaging/Dockerfile.mockbuild
Comment (frostedcmos) [reply packaging/Dockerfile.mockbuild:9]: Not sure I follow what is requested here. I've updated packaging/ due to previous PR complaining that packaging was out of date.

File: mercury.spec
Comment (brianjmurrell) [reply mercury.spec:16]: Are your tabs set to 4 spaces or something?  I see one tab after `Source0:` but two after `Patch:`.

Comment (frostedcmos) [reply mercury.spec:16]: no, it is set to 8. yes there are 2 tabs after patch,  same as 'group' line on line 11 that shows up misaligned in github rendering

Comment (brianjmurrell) [reply mercury.spec:16]: OK.

File: mercury.spec
Comment (frostedcmos) [reply mercury.spec:111]: After internal discussion between ofi/cart/mercury, it was decided to not include this patch

**Review by brianjmurrell — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

File: debian/changelog
Comment (brianjmurrell) [inline debian/changelog:5]: Could make the date here more accurate if you need to push again for any reason.

File: mercury.spec
Comment (brianjmurrell) [inline mercury.spec:110]: Ditto.

File: packaging/Dockerfile.mockbuild
Comment (brianjmurrell) [reply packaging/Dockerfile.mockbuild:9]: Simply checkout the bmurrell/leap15-build-should-be-15.3 branch in the packaging repo and update the packaging here from that.  All of the packaging/ changes in this PR are actually reverting the newest packaging changes in this repo which came from that branch.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Just need to use the bmurrell/leap15-build-should-be-15.3 branch for packaging/ since that is what is already in this PR and the update to packaging's master is actually reverting those newer changes.

That packaging branch has now seen enough exercise that it's going to land as soon as I get a minute to land it.

**Comment by frostedcmos** [comment]
DO NOT land this for now; we are investigating issue with mercury team with this RC release

**Comment by brianjmurrell** [comment]
> DO NOT land this for now; we are investigating issue with mercury team with this RC release

It needs a `packaging/` update before landing anyway, but indeed, I don't land until @daos-stack/release-engineering has been added as a reviewer for exactly this kind of situation.

File: packaging/Dockerfile.mockbuild
Comment (frostedcmos) [reply packaging/Dockerfile.mockbuild]: done. based on discussion in chat, updated packaging from the latest master as changes have just landed there

File: debian/changelog
Comment (frostedcmos) [reply debian/changelog]: done

**Review by jolivier23 — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**
